Fix: item selling errors and missing price initialization in stores

DispersiaRoleplay opened the pull request:
Type of change:
-  Bug fix (non-breaking change which fixes an issue)
-  New feature (non-breaking change which adds functionality)
-  Breaking change (fix or feature that would cause existing functionality to not work as expected)
-  This change requires a documentation update. 

Motive for This Pull Request:
- This PR fixes several crashes and usability issues in vorp_stores, making the script safer for players while ensuring server owners still see clear warnings when their configs are incomplete.
- Provide a brief explanation of why these changes are being proposed and what they aim to achieve.
- Prevent nil errors when selling items with empty inventory (both food and weapons).
- Replace silent console prints with proper in-game notifications using Core.NotifyObjective, localized across all supported languages.
- Fix crashes caused by missing sellprice or buyprice by adding safe fallbacks (or 0, or {}).
- Ensure random prices are properly initialized from config on resource start, so items never sell for $0.
- Add print warnings in console whenever config is missing or invalid, so admins know exactly where the mistake is.
- Explain the necessity of these changes and how they will impact the framework or its users.

Before fixes:
1) Grocery store:
- selling food with empty inventory → only F8 console log, no player feedback.

2) Gun store (Rhodes):
Error 1:
`SCRIPT ERROR: @vorp_stores/client/client.lua:407: bad argument #2 to 'format' (number expected, got nil)`
caused by missing sellprice.

Error 2:
`SCRIPT ERROR: @vorp_stores/client/client.lua:447: attempt to perform arithmetic on a nil value (field 'sellprice')`
- caused by uninitialized prices when trying to sell again.
- Random prices sometimes caused all items to sell for $0.

After fixes:
- Grocery store now shows proper notification to player (T.notSelectedItemToSell) instead of console print.
- Weapon store no longer crashes:
  - sellprice safely falls back to 0 in client menu (or 0).
  - Server initializes sellprice/buyprice only if valid (tonumber(randomprice)), preventing nil arithmetic.
  - Added or {} in callbacks to ensure safe iteration.
- Random prices load correctly on resource start, no more $0 bug.
- All messages are localized: English, Português_PT, Português_BR, Français, German, Spanish, Czech.
- Console still shows clear print warnings when configs are missing, so admins know exactly what needs fixing.

Please describe the tests you have conducted to verify your changes. Provide instructions so we can reproduce these tests. Also, list any relevant details for your test configuration.
-  Tested selling food with empty inventory → in-game notification shown, no crash.
-  Tested selling weapons with empty inventory → menu opens correctly, no crash, shows notification.
-  Tested selling items with missing sellprice (e.g. Trapper’s Knife) → server no longer crashes, price loads properly.
-  Tested random prices initialization on resource start → no $0 bug, correct prices pulled from config.
-  Verified all locales updated: EN, PT, FR, DE, ES, CZ.
-  Console logs warnings for missing configs, helping admins troubleshoot.
-  Tested with latest vorp scripts.
-  Tested with latest artifacts.

Notes if any
- These fixes improve stability for players (no more crashes, clear notifications).
- Server admins still see config mistakes via console warnings, so no errors are hidden.
- Fully backwards-compatible, safe to merge.

outsider31000 commented:
<img width="752" height="398" alt="image" src="https://github.com/user-attachments/assets/d1e508e4-7750-44fb-bda6-21b1aa2977ad" />

you are still hidding the errors here , if main config random prices were set then warn them when there's no random price in the sell or buy config.
also no need to convert a number to a number errors are essential to let people know what are they doing wrong.

<img width="1141" height="73" alt="image" src="https://github.com/user-attachments/assets/2e63b6ca-4bc9-4f07-a4f9-e4970120231a" />

Use what's in the documentation or like the rest of the script is using `Core.NotifyObjective`
once these are done i will merge

DispersiaRoleplay commented:
Thanks for the clear feedback

I’ve removed the unnecessary tonumber() conversion and added proper warnings when randomprice is missing, as you suggested.
Also switched to use Core.NotifyObjective for player notifications, consistent with the rest of the framework.

Should be good to go now

outsider31000 commented:
lets remove the notifications when something is wrong they have the prints it should be enough
<img width="1393" height="54" alt="image" src="https://github.com/user-attachments/assets/3a509ea8-983a-4bba-b4a3-362832eeaa15" />
